superdesk/io/commands/update_ingest.py

```diff
@@ -259,6 +259,28 @@ def process_anpa_category(item, provider):
         raise ProviderError.anpaError(ex, provider)
 
 
+def derive_category(item, provider):
+    """
+    Assuming that the item has at least one itpc subject use the vocabulary map to derive an anpa category
+    :param item:
+    :return: An item with a category if possible
+    """
+    try:
+        categories = []
+        subject_map = superdesk.get_resource_service('vocabularies').find_one(req=None, _id='iptc_category_map')
+        if subject_map:
+            for subject in item.get('subject', []):
+                for map_entry in subject_map['items']:
```

Comment:
I think it's better to move this to be the first iterator and filter for `is_active` there. Something like:
```
for entry in (for map_entry in subject_map['items'] if map_entry['is_active']):
     for subject in item.get('subject', []):
          ....
```